### DISCUSSION TITLE
video sensing

### DISCUSSION THREAD

**Discussion by dansrogers** [mKeRix/room-assistant#879] [Q&A] [answered]
Hi,

Any plans for camera based sensing?  A camera has wider angle of view, better resolution, lower cost, at the cost of more compute for objection detection -- and I already have an excess of compute in the raspberry pi module.

**Comment by sharpopotamus** [comment]
Object recognition is a whole other ballgame, you should check out: https://github.com/blakeblackshear/frigate. Frigate works well for person detection, although it's accuracy suffers at night and if you mount the cameras at too high an angle.

**Reply by dansrogers** [reply]
I agree it's a harder problem.  It would be nice though if I could incorporate it.  I'm ordering the thermopile sensors now and it seems like they have a pretty narrow field of view of 45 degrees for the 50 dollar one.  That means that at 8 ft ceiling installation you only get a 6.5 ft square on the ground. So I can install a cluster of four sensors for $200 but doing that per room is getting pretty crazy.  There are other possible mounting points but even with optimal placement a 15x30 ft room (my family room) at 6.5*6.5 sqft is 11 sensors or $550.  That's a lot.

OTOH frigate requires a minimum of 300 px by 300px image for each object.  For the 12.3 MP 4:3 raspberry pi camera, at 30 ft you get 150 pixels per foot.  That's enough to pick up a 1 year old or a pet at 30 ft.  So very likely 1 camera every 30 ft is sufficient.  One PI with one coral accelerator is enough for 3 or 4 cameras.  So 2 HQ cameras (50 bucks each), 1 pi zero and 1 pi 4 (for each camera) plus a coral unit on the beefier pi means $200 for comprehensive room coverage in my largest room, and I still have capacity for 2 more cameras on the one coral unit.

**Comment by mKeRix** [comment, marked as answer]
Cameras can be great, although it's not an avenue I have explored much because we had some reservations around being filmed while at home. They also have some issues around night time, although that can be worked around. I did actually order one of those OAK-D devices mentioned in #232, but I never got around to using it (mostly because there is no NodeJS library yet). Those would make it possible to run object detection even off the Pi Zero Ws, which is the board that I target most features of this software at.

Aside from that specific camera module, I think the way to go is a central computation unit that has the necessary machine learning power attached and the decentralized camera units on e.g. Pi Zero Ws. While I have nothing against integrating this into room-assistant, it would require quite a bit of work and I don't see myself having the time and energy to get work on this anytime soon (especially with so many other things open as well). I'm happy to help with PRs if anyone wants to pick this up though.

There might also be other software out there that can already do this, on the Pis it should be fairly easy to have multiple things running at the same time. I only know Frigate, which was already mentioned above, but I'm also not super aware of the market.
